From b5a48a1c32dbe41257efbdd4546721c1a03f338d Mon Sep 17 00:00:00 2001
From: Alexandre TORGUE <alexandre.torgue@st.com>
Date: Tue, 27 Feb 2024 13:10:36 +0100
Subject: [PATCH 602/841] feat(stm32mp25-fdts): update RIF configuration

Align the "open" configuration with the latest version of the xls tool.
It fixes the bl31_context area (start addr + size) and allows the CID0
to access to the "op-tee" area (as it is done for all area for a "open"
configuration).

It also changes PLL binding to satisfy YAML verification and moves
U(S)ART source clock to HSI_KER for low power feature.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git ebc7986ccfce28b519aaa0cea8006331b6a1401e feat(stm32mp25-fdts): update RIF configuration
d9b3c28dde757f7608d5b5e4e3d7d133e62eb2d3 feat(stm32mp25-fdts): update RIF configuration for DK board]
Signed-off-by: Alexandre TORGUE <alexandre.torgue@st.com>
Change-Id: Ie26db2489e6c7d761b36be6cb03dd5730f296e2e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/380737
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp257f-ev1-ca35tdcid-fw-config.dtsi | 18 +++++++++---------
 fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi       |  9 ++++-----
 2 files changed, 13 insertions(+), 14 deletions(-)

diff --git a/fdts/stm32mp257f-ev1-ca35tdcid-fw-config.dtsi b/fdts/stm32mp257f-ev1-ca35tdcid-fw-config.dtsi
index 3eedf590a..6e55ae64f 100644
--- a/fdts/stm32mp257f-ev1-ca35tdcid-fw-config.dtsi
+++ b/fdts/stm32mp257f-ev1-ca35tdcid-fw-config.dtsi
@@ -1,19 +1,19 @@
-// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
 /*
  * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
  */
 
 /*
  * STM32MP25 tf-a firmware config
- * Project : eco
- * Generated by XLmx tool version 2.2 - 6/1/2023 12:15:35 PM
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 2/27/2024 11:46:17 AM
  */
 
 / {
 	dtb-registry {
 		soc_fw-config {
-			load-address = <0x0 0x81ff0000>;
-			max-size = <0x10000>;
+			load-address = <0x0 0x81fc0000>;
+			max-size = <0x40000>;
 		};
 		tos_fw {
 			load-address = <0x0 0x82000000>;
@@ -22,13 +22,13 @@
 	};
 
 	st-mem-firewall {
-		bl31_context: bl31-context@81ff0000 {
-			reg = <0x0 0x81ff0000 0x0 0x10000>;
-			st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID1_BF, RIF_CID1_BF, RIF_CID1_BF, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		bl31_context: bl31-context@81fc0000 {
+			reg = <0x0 0x81fc0000 0x0 0x40000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
 		};
 		op_tee: op-tee@82000000 {
 			reg = <0x0 0x82000000 0x0 0x2000000>;
-			st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID1_BF, RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
 		};
 	};
 };
diff --git a/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi b/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
index b98033dc9..77fb413f3 100644
--- a/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
+++ b/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
 /*
  * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
  * Author: Loic Pallardy loic.pallardy@foss.st.com for STMicroelectronics.
@@ -6,8 +6,8 @@
 
 /*
  * STM32MP25 Clock tree device tree configuration
- * Project : eco
- * Generated by XLmx tool version 2.2 - 3/8/2023 2:34:26 PM
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 2/27/2024 11:46:16 AM
  */
 
 &clk_hse {
@@ -46,7 +46,7 @@
 		FLEXGEN_CFG(2, XBAR_SRC_PLL4, 0, 1)
 		FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
 		FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
-		FLEXGEN_CFG(8, XBAR_SRC_PLL4, 0, 11)
+		FLEXGEN_CFG(8, XBAR_SRC_HSI_KER, 0, 0)
 		FLEXGEN_CFG(48, XBAR_SRC_PLL5, 0, 3)
 		FLEXGEN_CFG(51, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(52, XBAR_SRC_PLL4, 0, 5)
@@ -68,7 +68,6 @@
 		};
 	};
 
-	/* DRAM clock = 2 * PLL2 clock */
 	pll2: st,pll-2 {
 		st,pll = <&pll2_cfg_600Mhz>;
 
-- 
2.47.2

